### PR TITLE
fixing missing data and some bugs

### LINKED ISSUE #124: Missing value error
There is a missing value error for in these 2 code snippets:
https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/master/_episodes/05-preprocessor.md#exercise-adding-more-preprocessor-steps
https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/master/_episodes/05-preprocessor.md#example

This is my recipe:

```yaml
# ESMValTool
# recipe_example.yml
---
documentation:
  description: Demonstrate basic ESMValTool example

  authors:
    - demora_lee
    - mueller_benjamin
    - swaminathan_ranjini

  maintainer:
    - demora_lee

  references:
    - demora2018gmd
    # Some plots also appear in ESMValTool paper 2.

  projects:
    - ukesm

datasets:
  - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Omon, ensemble: r1i1p1, start_year: 1859, end_year: 2005}

preprocessors:
  prep_timeseries: # For 0D fields
    annual_statistics:
      operator: mean
    area_statistics:
      operator: mean
    depth_integration:

diagnostics:
  # --------------------------------------------------
  # Time series diagnostics
  # --------------------------------------------------
  diag_timeseries_temperature:
    description: simple_time_series
    variables:
      timeseries_variable:
        short_name: thetaoga
        preprocessor: prep_timeseries
    scripts:
      timeseries_diag:
        script: ocean/diagnostic_timeseries.py
```

Stack trace:

```
2020-08-07 13:38:55,570 UTC [9014] ERROR   Program terminated abnormally, see stack trace below for more information
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_task.py", line 730, in _run_task
    output_files = task.run()
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_task.py", line 242, in run
    self.output_files = self._run(input_files)
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 428, in _run
    product.apply(step, self.debug)
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 296, in apply
    self.cubes = preprocess(self.cubes, step, **self.settings[step])
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 241, in preprocess
    result.append(_run_preproc_function(function, item, settings))
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 224, in _run_preproc_function
    return function(items, **kwargs)
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/_volume.py", line 307, in depth_integration
    depth = cube.coord(axis='z')
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/iris/cube.py", line 1497, in coord
    raise iris.exceptions.CoordinateNotFoundError(msg)
iris.exceptions.CoordinateNotFoundError: 'Expected to find exactly 1  coordinate, but found none.'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_main.py", line 430, in run
    fire.Fire(ESMValTool())
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/fire/core.py", line 138, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/fire/core.py", line 463, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/fire/core.py", line 672, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_main.py", line 407, in run
    process_recipe(recipe_file=recipe, config_user=cfg)
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_main.py", line 98, in process_recipe
    recipe.run()
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_recipe.py", line 1358, in run
    run_tasks(self.tasks,
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_task.py", line 650, in run_tasks
    _run_tasks_parallel(tasks, max_parallel_tasks)
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_task.py", line 695, in _run_tasks_parallel
    _copy_results(task, running[task])
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_task.py", line 718, in _copy_results
    task.output_files, updated_products = future.get()
  File "/home/stef/miniconda3/envs/esmvaltool/lib/python3.8/multiprocessing/pool.py", line 771, in get
    raise self._value
iris.exceptions.CoordinateNotFoundError: 'Expected to find exactly 1  coordinate, but found none.'

```

### PR DESCRIPTION
**Pull request by SarahAlidoost**
# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [x] Give this pull request a descriptive title.
- [x] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [x] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [x] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #124 #126 #127 #128 #129 #131 #99

### CODE REVIEW

File: data/dataset.urls [hunk 1]
```diff
@@ -47,3 +49,5 @@ http://crd-esgf-drc.ec.gc.ca/thredds/fileServer/esgC_dataroot/AR6/CMIP6/CMIP/CCC
 http://crd-esgf-drc.ec.gc.ca/thredds/fileServer/esgC_dataroot/AR6/CMIP6/CMIP/CCCma/CanESM5/historical/r2i1p2f1/Amon/tas/gn/v20190429/tas_Amon_CanESM5_historical_r2i1p2f1_gn_185001-201412.nc
 http://crd-esgf-drc.ec.gc.ca/thredds/fileServer/esgC_dataroot/AR6/CMIP6/CMIP/CCCma/CanESM5/historical/r3i1p2f1/Amon/tas/gn/v20190429/tas_Amon_CanESM5_historical_r3i1p2f1_gn_185001-201412.nc
 http://crd-esgf-drc.ec.gc.ca/thredds/fileServer/esgC_dataroot/AR6/CMIP6/CMIP/CCCma/CanESM5/historical/r4i1p2f1/Amon/tas/gn/v20190429/tas_Amon_CanESM5_historical_r4i1p2f1_gn_185001-201412.nc
+# Variable = pr, model = GPCP-SG, project= obs4mips
+https://dpesgf03.nccs.nasa.gov/thredds/fileServer/obs4MIPs/observations/NASA-GSFC/Obs-GPCP/GPCP/V2.2/atmos/pr/pr_GPCP-SG_L3_v2.2_197901-201312.nc
```
Comment (sverhoeven) [inline data/dataset.urls:53]: When running example (see #126), still get missing data error.

Seems `pr_GPCP-SG_L3_v2.2_197901-201312.nc` needs to be in `~/esmvaltool_tutorial/data/Tier1/GPCP-SG`. The wget command in the setup episode puts it in  `~/esmvaltool_tutorial/data/`. 

We need to put it in the right place somewhere.

Comment (sverhoeven) [reply data/dataset.urls:53]: After wget command we could add?

```shell
mkdir -p Tier1/GPCP-SG
mv pr_GPCP-SG_L3_v2.2_197901-201312.nc Tier1/GPCP-SG/
```

File: _episodes/05-preprocessor.md [hunk 2]
```diff
@@ -314,11 +341,8 @@ simple preprocessor and diagnostic setup for that:
 >         preprocessor: prep_regrid
 >         mip: Amon
 >         grid: gn  # can change for variables from the same model
->         start_year: 1970  # some 30 year period
+>         start_year: 1979
 >         end_year: 2000
->         additional_datasets:
```
Comment (sverhoeven) [inline _episodes/05-preprocessor.md:319]: Removal of observation data set should also be reflected in text. Now says we use observations in 2 vars. While with this change only pr has observations.

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Nice looking diff for recipe_example_thetao.yml.
This PR indeed solves #124 #127 #128 #129 #131 #99.
Was able to run all recipes in data/ successfully. 

Increased download size to 13Gb myself.

However explanation of reference (#127) could be improved, see suggestions.

First recipe in #126 still does not work, see comments in dataset.urls

File: _episodes/04-recipe.md [hunk 3]
```diff
@@ -181,14 +181,22 @@ Please note the following sections:
...
-      `esmvaltool/config-references.yml`)
+    - authors: a list of authors
+    - maintainer: a list of maintainers
+    - references: a list of references
+    - projects: a list of projects
 
+> ## config-references.yml and references
+>
+> The values for the keys ``author``, ``maintainer``, ``projects`` and
+> ``references`` in the recipe should be known by ESMValTool. A list of ESMValTool
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/04-recipe.md:192]: This part of the episode explains different sections in the r[ecipe_example](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/master/data/recipe_example.yml). the key "references" is mentioned in line 15 in this file.

File: _episodes/02-installation.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 title: "Installation"
 teaching: 10
-exercises: 20
+exercises: 10
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/02-installation.md:4]: Thanks. I see the points. However, here "exercises" is related to the challenge in the episode. There is only one challenge "Version of ESMValTool" at the end. So, I think 20 minutes is too long for this.

File: _episodes/04-recipe.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sverhoeven) [reply _episodes/04-recipe.md:192]: Doh, I should have read more carefully

File: _episodes/06-debugging.md [hunk 5]
```diff
@@ -215,18 +215,15 @@ attach the run/recipe_*.yml and run/main_log_debug.txt files from the output dir
 
 The values for the keys ``author``, ``maintainer``, ``projects`` and
 ``references`` in the recipe should be known by ESMValTool. A list of ESMValTool
```
[the remark this answers is not in the document]
Comment (sverhoeven) [reply _episodes/06-debugging.md:217]: Retracted, same as above

File: _episodes/02-installation.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sverhoeven) [reply _episodes/02-installation.md:4]: Cool

File: _episodes/06-debugging.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/06-debugging.md:217]: Again, this part of the episode explains different sections in the recipe_example. the key "references" is mentioned in line 15 in this file. I explained that  "A list of ESMValTool author, maintainer, and projects can be found in the config-references.yml." and "And ESMValTool references in BibTeX format can be found here."

File: data/dataset.urls (hunk 1 above)
Comment (SarahAlidoost) [reply data/dataset.urls:53]: @sverhoeven thanks. I added these lines.

File: _episodes/05-preprocessor.md (hunk 2 above)
Comment (SarahAlidoost) [reply _episodes/05-preprocessor.md:319]: it is fixed now.

File: setup.md [hunk 6]
```diff
@@ -284,6 +257,9 @@ To download the data, run the following command using [wget](https://en.wikipedi
 wget --no-clobber --input-file \
   https://github.com/ESMValGroup/ESMValTool_Tutorial/raw/master/data/dataset.urls \
   --directory-prefix $HOME/esmvaltool_tutorial/data/
+cd $HOME/esmvaltool_tutorial/data/
```
Comment (sverhoeven) [inline setup.md:260]: Could you explain why these (mkdir/mv) lines are needed in the setup.md? Maybe in separate text + code block

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Looks OK, Just a small nitpick about mkdir/mv

File: setup.md (hunk 6 above)
Comment (SarahAlidoost) [reply setup.md:267]: Thanks. I added some explanations.

Comment (sverhoeven) [reply setup.md:267]: great

**Review by sverhoeven — APPROVED** [review]
Everything looks OK, that is a lot of issues solved with a single PR.

**Comment by SarahAlidoost** [comment]
@sverhoeven Can I merge this PR?

**Pull request merged**